lib/api/apiUtils/bucket/getNotificationConfiguration.js

```diff
@@ -31,6 +31,45 @@ function getNotificationConfiguration(parsedXml) {
         error = error.addMetadataEntry('invalidArguments', invalidTargets);
         return { error };
     }
+
```

Comment:
There is already a validation in arsenal (if I recall correctly), where we traverse the whole list already.

Should the check not be added in there?

Reply:
arsenal checks against all possible events, just like aws does with a generic error for the whole payload.

Here the validation is specific to cloudserver config with specific message, just like it's done for the target validation

Reply:
yes, but I wonder if this validation should not be done as well by arsenal: i.e. keep related things together.

the arsenal function could be passed the "supported lifecycle rules", just like we do for validating putBucetLifecycleConfiguration.

Reply:
my point is not that we should handle update, but really that it seems a very valid use case to setup notification _before_ setting up the (cold) location.

generally, we should not restrict the API because _user_ has created some resources : but only because of _product_ configuration (i.e. something _we_ control, not the _end user_)

Reply:
Reworked in arsenal: https://github.com/scality/Arsenal/pull/2527